hs/src/Reach/Debugger.hs

```diff
@@ -0,0 +1,100 @@
+{-# OPTIONS_GHC -Wno-missing-export-lists #-}
+
+module Reach.Debugger where
+
+import Control.Monad.Reader
+
+import qualified Data.Map.Strict as M
+import Reach.AST.Base
+import Reach.AST.DLBase
+import Reach.AST.LL
+
+-- NOTE: let several base types be Integer/String as a placeholder for now
+
+type ConsensusEnv = M.Map DLVar DLVal
+
+type Balances = [Double]
```

Comment:
They won't be doubles... they will be naturals.

Reply:
Why a list? Each account would have just one, unless you're thinking of tokens. In which case, I'd say that the ledger would be `M.Map Account (M.Map Token Balance)`